clang/lib/Sema/SemaDecl.cpp

```diff
@@ -9360,15 +9360,16 @@ static FunctionDecl *CreateNewFunctionDecl(Sema &SemaRef, Declarator &D,
         SemaRef.Context, DC, D.getBeginLoc(), NameInfo, R, TInfo, SC,
         SemaRef.getCurFPFeatures().isFPConstrained(), isInline, HasPrototype,
         ConstexprSpecKind::Unspecified,
-        /*TrailingRequiresClause=*/nullptr);
+        /*TrailingRequiresClause=*/{});
     if (D.isInvalidType())
       NewFD->setInvalidDecl();
 
     return NewFD;
   }
 
   ExplicitSpecifier ExplicitSpecifier = D.getDeclSpec().getExplicitSpecifier();
-  Expr *TrailingRequiresClause = D.getTrailingRequiresClause();
+  AssociatedConstraint TrailingRequiresClause(
+      D.getTrailingRequiresClause(), /*ArgumentPackSubstitutionIndex=*/-1);
```

Comment:
This is REALLY getting overwhelming with the frequent assignment to `-1`. It means that we can never change the `no index` value without doing a massive audit here.  `AssociatedConstraint` should just have a `not an index` constructor (or, perhaps, a `::Create` function?) that just takes the Expr, so that all of the `-1` values are in the same place.